### PR TITLE
Refactored BC Player and Nested view

### PR DESCRIPTION
**Pull request by rafbc**

### CODE REVIEW

File: src/BCPlayer.js [hunk 1]
```diff
@@ -1,91 +1,149 @@
-import React, { Component } from 'react';
-import { StyleSheet, Platform } from 'react-native';
-import BrightcovePlayer from './BrightcovePlayer';
+import React, {Component} from 'react'
+import {Animated, BackHandler, Dimensions, Image, Platform, StatusBar, StyleSheet, Text} from 'react-native'
+import Orientation from 'react-native-orientation'
+import BrightcovePlayer from "./BrightcovePlayer";
+import PlayerEventTypes from "react-native-brightcove-player/src/PlayerEventTypes";
```
Comment (rafbc) [inline src/BCPlayer.js]: @pankaj-pp any reason why not just use
`import PlayerEventTypes from "./PlayerEventTypes";`

Comment (pankaj-pp) [reply src/BCPlayer.js]: Yes , my bad. Will fix it

File: src/BCPlayer.js [hunk 2]
```diff
@@ -1,91 +1,148 @@
...
+	componentWillUnmount() {
+		Dimensions.removeEventListener('change', this.onRotated)
+		BackHandler.removeEventListener('hardwareBackPress', this.BackHandler)
+	}
 
-		if (this.state.orientation === 'PORTRAIT') {
-			this.setState({ forcedOrientation: true });
-			Orientation.lockToLandscape();
+	onRotated({window: {width, height}}) {
+		if (this.props.inlineOnly) return
```
Comment (rafbc) [inline src/BCPlayer.js]: @pankaj-pp  i noticed that most of the lines doesn't end with ";". 
I would highly recommend to do so, as a good practice in Javascript.

Comment (pankaj-pp) [reply src/BCPlayer.js]: @rafbc yes, I'll do that. Actually we don't use semicolons in our projects, so that's the reason i missed it.

**Pull request merged**